integration_tests/commands/http/zset_test.go

```diff
@@ -105,3 +106,83 @@ func TestZPOPMIN(t *testing.T) {
+			expected: []interface{}{float64(0)}, // Expecting count of 0
+		},
+		{
+			name: "ZCOUNT on empty sorted set",
+			commands: []HTTPCommand{
+				{Command: "ZADD", Body: map[string]interface{}{"key": "myzset", "values": [...]string{"10", "member1"}}},
+				{Command: "ZCOUNT", Body: map[string]interface{}{"key": "myzset", "values": [...]string{"0", "5"}}},
+			},
+			expected: []interface{}{float64(1), float64(0)}, // Expecting ZADD to return 1, ZCOUNT to return 0
+		},
+		{
+			name: "ZCOUNT with invalid range",
+			commands: []HTTPCommand{
+				{Command: "ZADD", Body: map[string]interface{}{"key": "myzset", "values": [...]string{"10", "member1"}}},
+				{Command: "ZCOUNT", Body: map[string]interface{}{"key": "myzset", "values": [...]string{"20", "5"}}},
+			},
+			expected: []interface{}{float64(1), float64(0)}, // Expecting ZADD to return 1, ZCOUNT to return 0
+		},
+		{
+			name: "ZCOUNT with valid key and range",
+			commands: []HTTPCommand{
+				{Command: "ZADD", Body: map[string]interface{}{"key": "myzset", "values": [...]string{"10", "member1", "20", "member2", "30", "member3"}}},
+				{Command: "ZCOUNT", Body: map[string]interface{}{"key": "myzset", "values": [...]string{"15", "25"}}},
+			},
+			expected: []interface{}{float64(3), float64(1)}, // Expecting ZADD to return 3, ZCOUNT to return 1
+		},
+		{
+			name: "ZCOUNT with min and max values outside existing members",
+			commands: []HTTPCommand{
+				{Command: "ZADD", Body: map[string]interface{}{"key": "myzset", "values": [...]string{"5", "member1", "15", "member2", "25", "member3"}}},
+				{Command: "ZCOUNT", Body: map[string]interface{}{"key": "myzset", "values": [...]string{"30", "50"}}},
+			},
+			expected: []interface{}{float64(3), float64(0)}, // Expecting ZADD to return 3, ZCOUNT to return 0
+		},
+		{
+			name: "ZCOUNT with multiple members having the same score",
+			commands: []HTTPCommand{
+				{Command: "ZADD", Body: map[string]interface{}{"key": "myzset", "values": [...]string{"10", "member1", "10", "member2", "10", "member3"}}},
+				{Command: "ZCOUNT", Body: map[string]interface{}{"key": "myzset", "values": [...]string{"0", "20"}}},
+			},
+			expected: []interface{}{float64(3), float64(3)}, // Expecting ZADD to return 3, ZCOUNT to return 3
+		},
+		{
+			name: "ZCOUNT with count argument exceeding the number of members",
+			commands: []HTTPCommand{
+				{Command: "ZADD", Body: map[string]interface{}{"key": "myzset", "values": [...]string{"1", "member1", "2", "member2"}}},
+				{Command: "ZCOUNT", Body: map[string]interface{}{"key": "myzset", "values": [...]string{"3", "10"}}},
+			},
+			expected: []interface{}{float64(2), float64(0)}, // Expecting ZADD to return 2, ZCOUNT to return 0
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			exec.FireCommand(HTTPCommand{
+				Command: "DEL",
+				Body:    map[string]interface{}{"key": "myzset"},
+			})
+			for i, cmd := range tc.commands {
+				fmt.Println("Checking zpop", cmd)
```

Comment:
```suggestion
```

Reply:
This was by mistake i was debugging but it's not there anymore